Fix for bug #44: Symbolic toolbar icons and code refactoring

jakubbrindza opened the pull request:
I have worked on implementing nicer and more representative icons for the current design of GTG.
The perviously used icons were very similar and caused confusion. This should become past with these GTK3 icons.

This applies not only to toolbar icons, but also to the right-click menu icons for corresponding actions.

Previous icons:
![gtg-before](https://cloud.githubusercontent.com/assets/2866323/6750144/428c6752-ceef-11e4-83b2-95326798526d.png)

Changed into:
![gtg-after](https://cloud.githubusercontent.com/assets/2866323/6750146/428ee7fc-ceef-11e4-8502-f37cd97fde8f.png)

Some other new stuff in cancelled tasks:
Before (Done task):
![gtg_before_cancel_edit](https://cloud.githubusercontent.com/assets/2866323/6750145/428e7858-ceef-11e4-9333-953bc6a59727.png)

After (Done task):
![gtg_after_cancel_edit](https://cloud.githubusercontent.com/assets/2866323/6750143/428b3c9c-ceef-11e4-8953-7076ef1bf618.png)

Before (Dismissed task):
![gtg_before_cancel_edit_2](https://cloud.githubusercontent.com/assets/2866323/6750184/b23533c2-ceef-11e4-886b-3765e7acf0c8.png)

After (Dismissed task):
![gtg_after_closed_edit_2](https://cloud.githubusercontent.com/assets/2866323/6750190/c06d87b4-ceef-11e4-9666-c01945d42731.png)

izidormatusov commented:
Icon for dismissing tasks looks too much similar as icon to delete tasks. Could you experiment if you would have a better result with `Gtk.STOCK_CLOSE` or `Gtk.STOCK_DISCARD`? Can you make the icon when a dimsissed task is selected to be `Gtk.STOCK_REFRESH` as well? It would symbolise putting it back in the work queue. 

There is opportunity to clean GTG from unused icons. Checklist of things to do:

 - [ ] change icons also in GTG/gtk/browser/taskbrowser.ui
   - [ ] gtg-task-done
   - [ ] gtg-task-dismiss
   - [ ] gtg-task-new
 - [ ] remove following icon files:

```
./data/icons/hicolor/16x16/actions/gtg-task-done.png
./data/icons/hicolor/22x22/actions/gtg-task-done.png
./data/icons/hicolor/16x16/actions/gtg-task-undone.png
./data/icons/hicolor/22x22/actions/gtg-task-undone.png
./data/icons/hicolor/16x16/actions/gtg-task-dismiss.png
./data/icons/hicolor/22x22/actions/gtg-task-dismiss.png
./data/icons/hicolor/16x16/actions/gtg-task-undismiss.png
./data/icons/hicolor/22x22/actions/gtg-task-undismiss.png
./data/icons/hicolor/16x16/actions/gtg-task-new.png
./data/icons/hicolor/22x22/actions/gtg-task-new.png

jakubbrindza commented:
Changes of the DISMISS button into a more distinct button (see the picture). I also deleted the unused icons, this should then solve the bug entirely.

![gtg_after_final](https://cloud.githubusercontent.com/assets/2866323/6771957/6e1941a2-d0e7-11e4-8709-ff41b588deb5.png)

jakubbrindza commented:
Changes to the DISMISS button which is now also more distinct [see the pic] and unified undismiss and undo task buttons to represent the idea of "REFRESHING" them.
I also changed the old gtg-...-... icons in the code for their GTK3 equivalents and removed them from the repo completely, since they are not used and will not be used any more.

![gtg_after_final](https://cloud.githubusercontent.com/assets/2866323/6772043/52148acc-d0ea-11e4-848a-c0ba87959ba4.png)
![gtg_after undismiss](https://cloud.githubusercontent.com/assets/2866323/6772048/77b17682-d0ea-11e4-948a-e180339a6060.png)
![gtg-after not done](https://cloud.githubusercontent.com/assets/2866323/6772049/77dbda30-d0ea-11e4-8908-931c5a657247.png)

izidormatusov commented on GTG/gtk/browser/browser.py:
```diff
@@ -1305,8 +1305,8 @@ def on_task_cursor_changed(self, selection=None):
         # Only if something is selected in the active task list
-        self.donebutton.set_icon_name("gtg-task-done")
-        self.dismissbutton.set_icon_name("gtg-task-dismiss")
+        self.donebutton.set_icon_name(Gtk.STOCK_APPLY)
```
Above you use set_stock_id for the button, down here you use set_icon_name. Could you use just single way how to set buttons? Is there a difference between those calls?

izidormatusov commented on GTG/gtk/browser/taskbrowser.ui:
```diff
@@ -397,7 +397,7 @@
                     <property name="can_focus">False</property>
                     <property name="label" translatable="yes">New Subtask</property>
-                    <property name="icon_name">gtg-task-new</property>
+                    <property name="icon_name">Gtk.STOCK_NEW</property>
```
I don't think this is going to work, you need to use the literal value here: `gtk-new`

izidormatusov commented on GTG/gtk/browser/taskbrowser.ui:
```diff
@@ -809,25 +809,25 @@ Bertrand Rousseau (bertrand.rousseau@gmail.com)</property>
     <property name="can_focus">False</property>
     <property name="pixel_size">16</property>
-    <property name="icon_name">gtg-task-dismiss</property>
+    <property name="icon_name">Gtk.STOCK_CLOSE</property>
```
same as above

izidormatusov commented on GTG/gtk/browser/taskbrowser.ui:
```diff
@@ -809,25 +809,25 @@ Bertrand Rousseau (bertrand.rousseau@gmail.com)</property>
     <property name="can_focus">False</property>
     <property name="pixel_size">16</property>
-    <property name="icon_name">gtg-task-done</property>
+    <property name="icon_name">Gtk.STOCK_APPLY</property>
```
same as above

izidormatusov commented on GTG/gtk/browser/taskbrowser.ui:
```diff
@@ -809,25 +809,25 @@ Bertrand Rousseau (bertrand.rousseau@gmail.com)</property>
     <property name="can_focus">False</property>
     <property name="pixel_size">16</property>
-    <property name="icon_name">gtg-task-new</property>
+    <property name="icon_name">Gtk.STOCK_NEW</property>
```
same as above

izidormatusov commented on GTG/gtk/browser/taskbrowser.ui:
```diff
@@ -809,25 +809,25 @@ Bertrand Rousseau (bertrand.rousseau@gmail.com)</property>
     <property name="can_focus">False</property>
     <property name="pixel_size">16</property>
-    <property name="icon_name">gtg-task-new</property>
+    <property name="icon_name">Gtk.STOCK_NEW</property>
```
same as above

izidormatusov commented on GTG/gtk/editor/editor.py:
```diff
@@ -258,24 +258,24 @@ def refresh_editor(self, title=None, refreshtext=False):
             self.dismissbutton.set_label(GnomeConfig.MARK_UNDISMISS)
             self.dismissbutton.set_tooltip_text(undismiss_tooltip)
-            self.dismissbutton.set_icon_name("gtg-task-undismiss")
+            self.dismissbutton.set_stock_id(Gtk.STOCK_REFRESH)
```
Again, why to combine set_icon_name and set_stock_id for the same purpose? Pick single way how to set icons. Consistency makes the code much readable

izidormatusov commented:
Please, fix the remarks

izidormatusov commented on GTG/gtk/browser/browser.py:
```diff
@@ -1242,89 +1242,84 @@ def on_select_tag(self, widget=None, row=None, col=None):
+        If no tasks at all are selected, set status to STA_ACTIVE"""
+
+        if selection.count_selected_rows() > 0 and
+        'closed' in self.vtree_panes:
```
Can you indent it by 2x4 spaces to make sure that it is visually still part of the condition?

izidormatusov commented:
Congratulation on your first contribution!
